process/utility/utils.go

```diff
@@ -2,26 +2,32 @@ package utility
 
 import (
 	"bytes"
+	"encoding/hex"
+	"fmt"
 	"math/big"
 
 	"github.com/ElrondNetwork/elrond-go-core/core"
 	"github.com/elodina/go-avro"
 )
 
-// StrSliceToBytesSlice outputs the bytes slice representation of a string slice input
-func StrSliceToBytesSlice(in []string) [][]byte {
+// HexSliceToByteSlice outputs a decoded byte slice representation of a hex string encoded slice input
+func HexSliceToByteSlice(in []string) ([][]byte, error) {
 	if in == nil {
-		return nil
+		return nil, nil
 	}
-
 	out := make([][]byte, len(in))
 
 	for i := range in {
-		out[i] = make([]byte, len(in[i]))
-		out[i] = []byte(in[i])
+		tmp, err := hex.DecodeString(in[i])
+		if err != nil {
+			return nil, err
+		}
+
+		out[i] = make([]byte, len(in[i])/2)
```

Comment:
I would remove this line.
There is no need to sett in out[i] a pointer to a newly allocated slice, since you replace then the pointer with tmp.

Reply:
Yes, good point. I left it there thinking it would could lead to some **uninitialized** memory, but this cannot happen actually.
Fixed it